### PR TITLE
DAOS-2236 utils: add target exclude to dmg exclude

### PR DESCRIPTION
**Pull request by wangdi1**
Add single target exclude to dmg exclude command.

Signed-off-by: Wang Di <di.wang@intel.com>

### CODE REVIEW

File: src/utils/dmg.c [hunk 1]
```diff
@@ -378,10 +387,16 @@ pool_op_hdlr(int argc, char *argv[])
...
+		D_ASSERT(ranks->rl_nr == 1);
 		tgt_list.tl_nr = 1;
-		tgt_list.tl_ranks = targets->rl_ranks;
-		tgt_list.tl_tgts = &tgt;
+		tgt_list.tl_ranks = ranks->rl_ranks;
+		if (targets != NULL) {
+			D_ASSERT(targets->rl_nr == 1);
+			tgt_list.tl_tgts = (int *)targets->rl_ranks;
+		} else {
+			tgt_list.tl_tgts = &tgt;
```
Comment (okulkarni) [inline src/utils/dmg.c]: Where are we setting the value for `tgt`?

File: src/utils/dmg.c [hunk 2]
```diff
@@ -345,29 +352,31 @@ pool_op_hdlr(int argc, char *argv[])
...
+		ranks = daos_rank_list_parse(rank_str, ":");
+		if (ranks == NULL) {
+			fprintf(stderr, "failed to parse ranks\n");
+			return 2;
 		}
 	}
 
 	/* Check the targets for POOL_EXCLUDE, REPLICA_ADD & REPLICA_DEL. */
-	if (targets == NULL &&
+	if (ranks == NULL &&
```
Comment (okulkarni) [inline src/utils/dmg.c]: `ranks` is never really being used for replica_add or replica_del, right?

File: src/utils/dmg.c (hunk 1 above)
Comment (wangdi1) [reply src/utils/dmg.c]: tgt = -1 in this case means all targets will on this rank will be excluded.

File: src/utils/dmg.c (hunk 2 above)
Comment (wangdi1) [reply src/utils/dmg.c]: Ah, it should use ranks in replica_add and del. Thanks.

File: src/utils/dmg.c [hunk 3]
```diff
@@ -378,10 +387,16 @@ pool_op_hdlr(int argc, char *argv[])
...
 		/* Only support exclude single target XXX */
-		D_ASSERT(targets->rl_nr == 1);
+		D_ASSERT(ranks->rl_nr == 1);
 		tgt_list.tl_nr = 1;
-		tgt_list.tl_ranks = targets->rl_ranks;
-		tgt_list.tl_tgts = &tgt;
+		tgt_list.tl_ranks = ranks->rl_ranks;
+		if (targets != NULL) {
+			D_ASSERT(targets->rl_nr == 1);
+			tgt_list.tl_tgts = (int *)targets->rl_ranks;
```
Comment (gnailzenh) [inline src/utils/dmg.c]: It is a little confusing to reuse the rank_list for target at here...

**Pull request merged**